cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/store/AppMetadataStore.java

```diff
@@ -448,23 +571,32 @@ public boolean apply(RunRecordMeta input) {
     return getNonCompleteRuns(programRunIds, TYPE_RUN_RECORD_STARTED, limit);
   }
 
-  private Map<ProgramRunId, RunRecordMeta> getNonCompleteRuns(Set<ProgramRunId> programRunIds, String recordType,
+  private Map<ProgramRunId, RunRecordMeta> getHistoricalRuns(Set<ProgramRunId> programRunIds, int limit) {
+    return getNonCompleteRuns(programRunIds, TYPE_RUN_RECORD_COMPLETED, limit);
```

Comment:
Historical runs are completed runs, using `getNonCompleteRuns()` to get historical runs looks inconsistent. It is better to call this method as `getRunsForRunIds()`